core/src/tile/tileID.h

```diff
@@ -41,14 +44,30 @@ struct TileID {
         return isValid() && z <= _maxZoom;
     }
 
+    TileID withMaxSourceZoom(int32_t _max) const {
```

Comment:
Not sure `withMaxSourceZoom` is the appropriate name here?
how about `tileIDUnzoomed` or something on those lines.